File: src/sdk-reference/android/3/profile/update/snippets/update-1.java

JSONObject policy1 = new JSONObject().put("roleId", "myrole");

JSONObject updateContent = new JSONObject()
  .put("policies", new JSONArray().put(policy1));

profile.update(updateContent, new ResponseListener<Profile>() {
  @Override
  public void onSuccess(Profile updatedProfile) {

  }

  @Override
  public void onError(JSONObject error) {

  }
});

File: src/sdk/php/3/core-classes/profile/update/snippets/update-1.php

<?php

use Kuzzle\Security\Profile;

// ...

/*
 * @var $profile Profile
 */
$profileContent = [
  'policies' => [
    ['roleId' => 'myrole']
  ]
];

try {
  $profile->update($profileContent);
}
catch (ErrorException $e) {

}
